contributors/design-proposals/bootstrap-discovery.md

```diff
@@ -173,6 +171,44 @@ In addition, `jws-kubeconfig-<token-id>-hash` will be set to the MD5 hash of the
 
 This controller will also delete tokens that are past their expiration time.
 
+## `kubeadm` UX
+
+We extend kubeadm with a set of flags and helper commands for managing and using these tokens.
+
+### `kubeadm init` flags
+
+* `--token` If set, this injects the bootstrap token to use when initializing the cluster.  If this is unset, then a random token is created and shown to the user.  If set explicitly to the empty string then no token is generated or created.
+* `--token-ttl` If set, this sets the TTL for the lifetime of this token.  Defaults to 0 which means "forever"
+* `--discovery-token` If set, this is a token that is created in the new cluster that is used for signing discovery bootstrap.  If not set, it defaults to the value from `--token`.  If set to the empty string, there is no token created.
+* `--discovery-token-ttl` Similar TTL for token.
+* `--bootstrap-auth-token` (not strictly part of this proposal) If set, this token is set into the cluster to be used for doing authentication for TLS bootstrap.  If set to the empty string there is no token created.
```

Comment:
So -- do we want to support having a different token for different usages?  I can see going down to a single token but I want to make sure we don't preclude scenarios here.